staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -408,6 +407,7 @@ func (s *sharedIndexInformer) Run(stopCh <-chan struct{}) {
 		s.stopped = true // Don't want any new listeners
 	}()
 	s.controller.Run(stopCh)
+	wg.Wait() // Wait for Processor to stop
```

Comment:
This is a little worrying. "defer"s are LIFO, so you are changing the relative order of setting stopped and calling wait. I suspect we are adding a potential race condition with this change.

Reply:
Good spot.  The ordering with `close(processorStopCh)` is probably also important.

It was actually only `controller.Run()` that went wrong in real life, and I see #93646 has now changed to adopt my fix for that , so maybe you should accept that.